clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -461,7 +463,7 @@ private void handshakeFinished() throws IOException {
             if (netWriteBuffer.hasRemaining())
                 key.interestOps(key.interestOps() | SelectionKey.OP_WRITE);
             else {
-                state = sslEngine.getSession().getProtocol().equals("TLSv1.3") ? State.POST_HANDSHAKE : State.READY;
+                state = sslEngine.getSession().getProtocol().equals(TLS13) ? State.POST_HANDSHAKE : State.READY;
```

Comment:
I don't know how often TLS gets a version update (guessing _extremely_ infrequently), but this feels fragile in the sense that the POST_HANDSHAKE part is likely to be relevant to 1.3 and later versions.

No action needed, but if I thought it likely we'd continue running this code long enough to see another TLS version, I'd ask for a warning comment at minimum, if not some kind of string-to-floating-point-version conversion and range check (though that's fragile as well).